src/pybind/Makefile

```diff
@@ -121,7 +121,7 @@ test: all
 	make -C chain test
 	make -C cudamatrix test
 	make -C dlpack test
-	$(eval include ../../tools/env.sh)
+	$(eval -include ../../tools/env.sh)
```

Comment:
I am afraid this does not work.

The error message I encountered is

```bash
$ make test
python3 tests/test_dlpack_subvector.py
make: python3: Command not found
Makefile:119: recipe for target 'test' failed
make: *** [test] Error 127
```

The file `../../tools/env.sh` always exists, where you use MKL or not.

You should change `env.sh`

1. Use `:=` instead of `=`
2. Replace `$PATH` with `${PATH}`

Reply:
`env.sh` does not always exist. And the error message you show seems irrelevent. I test this fix with or without MKL install, both passed.

Reply:
See in [this line](https://github.com/kaldi-asr/kaldi/commit/277bbafab3304eb23a1950ce385579c33d1080dc#diff-8c42b80642e8e01f0c92a912760fd5beR259) that it is `:=` not `=`

Reply:
What I meant was `PATH`, NOT `LD_PRELOAD`.

Although you did not change `PATH`, but you included it to `Makefile`, `$PATH`
is not valid in `Makefile`, you have to use `${PATH}`.

Reply:
Double confirmed, even if `env.sh` is not clean, i.e. you have previous installation which generated `env.sh` with `=`, it won't raise any error regardless of you compile against IntelMKL or not.

Reply:
Could you paste your `env.sh`?

Reply:
`$PATH` is not used in the IntelMKL export, so it does not make a difference whether it is valid in `Makefile` or not.

Reply:
I see.
My `env.sh` is different from yours!

I have installed kaldi lm and this line caused the error for me:

https://github.com/kaldi-asr/kaldi/blob/752df8c700abf7a5488c07f4025fe18c27e244d3/tools/extras/install_kaldi_lm.sh#L31

I guess you only have one line `env.sh` that is why `=` is OK. But I have two lines.

`:=` is not valid in bash; but if you have multiple lines in `env.sh`, `=` is a lazy assignment,
I guess it is an infinite recursive substitution for `PATH`.